caikit/runtime/service_generation/data_stream_source.py

```diff
@@ -288,10 +303,15 @@ def make_data_stream_source(data_element_type: Type) -> Type[DataBase]:
 
         # Add an init that sequences the initialization so that
         # DataStreamSourceBase is initialized after DataBase
-        orig_init = _DataObjectBaseMetaClass._make_init(data_object.fields)
+        orig_init = _make_oneof_init(data_object)
 
         def __init__(self, *args, **kwargs):
-            orig_init(self, *args, **kwargs)
+            try:
+                orig_init(self, *args, **kwargs)
+            except TypeError as err:
+                raise CaikitRuntimeException(
```

Comment:
Since the `oneof_init` now does duplicate arg checking, the `CaikitRuntimeException` needed to be hoisted to here